.github/workflows/jirabot.yml

```diff
@@ -44,49 +43,56 @@ jobs:
                     print('Error: Unable to find transition for status: ' + statusName)
                 elif transition != '':
                     try:
-                        jira.transition_issue(issue, transition)
+                        jira.issue_transition(issueName, transition)
                         result += 'Workflow Transition: ' + transition + '\n'
                     except Exception as error:
-                        transitions = jira.transitions(issue)
+                        transitions = jira.get_issue_transitions(issueName)
                         result += 'Error: Transition: "' + transition + '" failed with: "' + str(error) + '" Valid transitions=' + str(transitions) + '\n'
 
                 prFieldName = propertyMap.get('pullRequestFieldName', 'customfield_10010')
-                try:
-                    currentPR = getattr(issue.fields, prFieldName)
-                except:
+
+                if prFieldName in issueFields:
+                    currentPR = issueFields[prFieldName]
+                else:
+                    print('Error: Unable to find pull request field with field name: ' + prFieldName)
                     currentPR = None
-                    print('Error: Unable to get current pull request with field name: ' + prFieldName)
 
                 if currentPR is None:
-                    issue.update(fields={prFieldName: pull_url})
+                    jira.update_issue_field(issueName, {prFieldName: pull_url})
                     result += 'Updated PR\n'
                 elif currentPR is not None and currentPR != pull_url:
                     result += 'Additional PR: ' + pull_url + '\n'
 
-                if prAuthorEmail:
-                    if issue.fields.assignee is None:
-                        jira.assign_issue(issue, prAuthorEmail)
-                        result += 'Assigning user: ' + prAuthorEmail + '\n'
-                    else:
-                        assigneeEmail = None
-                        if issue.fields.assignee:
-                            assigneeEmail = issue.fields.assignee.emailAddress
-                        if assigneeEmail is None or assigneeEmail.lower() != assigneeEmail.lower():
-                            result += 'Changing assignee from: ' + assigneeEmail + ' to: ' + prAuthorEmail + '\n'
-                            jira.assign_issue(issue, prAuthorEmail)
+                if prAuthor:
+                    assignee = issueFields['assignee']
+                    assigneeId = assignee['accountId']
+                    assigneeEmail = assignee["emailAddress"]
+
+                    prAuthorId = prAuthor["accountId"]
+                    prAuthorEmail = prAuthor["emailAddress"]
+                    if assigneeId is None or assigneeId == '':
+                        jira.assign_issue(issueName, prAuthorId)
+                        result += 'Assigning user: ' + prAuthor + '\n'
+                    elif assigneeId != prAuthorId:
```

Comment:
Not quite sure about this - failing currently helps catch PRs with the wrong ID.  Will merge as-is and revisit after some use.